### PR TITLE
Can now load an orebox onto the gigadrill

### PR DESCRIPTION
**Pull request by MadmanMartian**
Suggested by a random miner when I asked if they wanted anything added to mining.

Can now drag an ore box onto the gigadrill, where when you mine it autoloads resources into the ore box.

Drag off to offload the ore box.

🆑 
 * rscadd: You can now load an orebox onto the gigadrill so it collects resources as you mine with it. Offload the orebox by clickdragging to a nearby turf.

### CODE REVIEW

File: code/game/objects/structures/vehicles/gigadrill.dm [hunk 1]
```diff
@@ -63,6 +70,51 @@
...
 		M.GetDrilled()
+		if(OB)
+			var/count = 0
+			for(var/obj/item/weapon/ore/ore in range(src,1))
+				if(get_dir(src,ore)&dir && ore.material)
+					OB.materials.addAmount(ore.material,1)
+					returnToPool(ore)
+					count++
+			if(count)
+				to_chat(occupant,"<font color='blue'>[count] ore successfully loaded into cargo compartment.</font>")
```
Comment (jknpj) [inline code/game/objects/structures/vehicles/gigadrill.dm:81]: I don't know if this is a good idea

File: code/game/objects/structures/vehicles/gigadrill.dm [hunk 2]
```diff
@@ -11,6 +11,13 @@
 	keytype = /obj/item/key/gigadrill
 	wreckage_type = /obj/effect/decal/mecha_wreckage/vehicle/gigadrill
 	var/turf/drilling_turf
+	var/obj/structure/ore_box/OB //The orebox contained within
+
+/obj/structure/bed/chair/vehicle/gigadrill/destroy()
```
Comment (jknpj) [inline code/game/objects/structures/vehicles/gigadrill.dm]: Should be fine even without it, but 🤷‍♂️

Comment (jknpj) [reply code/game/objects/structures/vehicles/gigadrill.dm]: wait no, i'm wrong.

File: code/game/objects/structures/vehicles/gigadrill.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/game/objects/structures/vehicles/gigadrill.dm:81]: it's what the working mechs do

Comment (jknpj) [reply code/game/objects/structures/vehicles/gigadrill.dm:81]: Seems spammy as fuack

Comment (MadmanMartian) [reply code/game/objects/structures/vehicles/gigadrill.dm:81]: 🤷‍♂️ User feedback is user feedback. It's just telling you there was a success in loading some of the ores

Comment (Exxion) [reply code/game/objects/structures/vehicles/gigadrill.dm:81]: As one of like four people who ever mines with a mech, it's not as bad as you're imagining, @jknpj

**Pull request merged**